### PR TITLE
Added S1541 (function complexity rule) for VB.NET

### PR DESCRIPTION
**Pull request by erik-am**
For C# there already was a rule that gave a warning on high cyclomatic complexity, but this rule did not exist for VB.NET. This pull request adds a similar rule for VB.NET, re-using the same logic for calculating cyclomatic complexity that is also used for C#.

Functions, subprocedures and properties will be analysed.

I also added some test scenarios to the unit tests.

I hope this functionality can be incorporated into SonarLint!

### CODE REVIEW

File: src/SonarLint.VisualBasic/Rules/FunctionComplexity.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+using System;
```
Comment (tvsonar) [inline src/SonarLint.VisualBasic/Rules/FunctionComplexity.cs]: An empty new line is missing here

**Comment by tvsonar** [comment]
@erik-am The functionality of this PR looks good to me, but we try to merge implementation details of rules that are implemented in both C# and VB.NET. Have a look at https://github.com/SonarSource-VisualStudio/sonaranalyzer-csharp/tree/master/src/SonarLint/Rules.

Do you have the bandwidth to rework this PR to match that structure?

**Comment by erik-am** [comment]
Good to know that there are base classes for the common parts, I missed that! I intend to change it before Monday.

**Comment by tvsonar** [comment]
@erik-am Have you made any progress on this PR? In the meantime we decided that we're going to have a dedicated VB.NET sprint in the upcoming 3 weeks, so I think it makes more sense to include this new rule there.

File: src/SonarLint.VisualBasic/Rules/FunctionComplexity.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        private const int DefaultValueMaximum = 10;
+
+        [RuleParameter("maximumFunctionComplexityThreshold", PropertyType.Integer,
+            "The maximum authorized complexity in function", DefaultValueMaximum)]
+        public int Maximum { get; set; } = DefaultValueMaximum;
+
+        protected override void Initialize(ParameterLoadingAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
```
Comment (tvsonar) [inline src/SonarLint.VisualBasic/Rules/FunctionComplexity.cs]: Constructors, operators should also be handled here

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR sonaranalyzer-csharp-vbnet:sonaranalyzer-csharp-vbnet:D1AE804F-AE78-4883-B3D6-9E3C4026DEF6: Because field 'UnarySymbolicValue.operand' is visible outside of its declaring type, change its accessibility to private and add a property, with the same accessibility as the field has currently, to provide access to it. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=fxcop%3ADoNotDeclareVisibleInstanceFields)

**Comment by erik-am** [comment]
@tvsonar I implemented your suggestions.

**Comment by tvsonar** [comment]
Thanks @erik-am, I merged this PR through https://github.com/SonarSource-VisualStudio/sonaranalyzer-dotnet/commit/0a676a32d7035da626f0855faa79d3ee52e8fa8c .

**Pull request closed without merging**